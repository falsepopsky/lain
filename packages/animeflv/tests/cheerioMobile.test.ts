import { MobileBQ, MobileBQPT, MobileBrowse, MobileFake } from '../fixtures/index.js';
import { getPages, getTitlesMobile } from '../src/index.js';

describe('getTitlesMobile()', () => {
  describe('Fixture browse', () => {
    const titles = getTitlesMobile(MobileBrowse);

    it('Should return titles', () => {
      expect(Array.isArray(titles)).toBe(true);
      expect(titles).toHaveLength(24);
      expect(titles).toMatchSnapshot();
    });

    it('Should return a string type for every value of the keys from titles', () => {
      for (const title of titles) {
        expect(typeof title.title).toBe('string');
        expect(typeof title.slug).toBe('string');
        expect(typeof title.type).toBe('string');
        expect(typeof title.cover).toBe('string');
      }
    });

    it('Should have the expected keys in the data object', () => {
      for (const title of titles) {
        expect(title).toHaveProperty('title');
        expect(title).toHaveProperty('slug');
        expect(title).toHaveProperty('type');
        expect(title).toHaveProperty('cover');
      }
    });
  });

  describe('Fixture browse-query', () => {
    const titles = getTitlesMobile(MobileBQ);

    it('Should return titles', () => {
      expect(Array.isArray(titles)).toBe(true);
      expect(titles).toHaveLength(9);
      expect(titles).toMatchSnapshot();
    });

    it('Should return a string type for every value of the keys from titles', () => {
      for (const title of titles) {
        expect(typeof title.title).toBe('string');
        expect(typeof title.slug).toBe('string');
        expect(typeof title.type).toBe('string');
        expect(typeof title.cover).toBe('string');
      }
    });

    it('Should have the expected keys in the data object', () => {
      for (const title of titles) {
        expect(title).toHaveProperty('title');
        expect(title).toHaveProperty('slug');
        expect(title).toHaveProperty('type');
        expect(title).toHaveProperty('cover');
      }
    });
  });

  describe('Fixture browse-query-page-10', () => {
    const titles = getTitlesMobile(MobileBQPT);

    it('Should return titles', () => {
      expect(Array.isArray(titles)).toBe(true);
      expect(titles).toHaveLength(24);
      expect(titles).toMatchSnapshot();
    });

    it('Should return a string type for every value of the keys from titles', () => {
      for (const title of titles) {
        expect(typeof title.title).toBe('string');
        expect(typeof title.slug).toBe('string');
        expect(typeof title.type).toBe('string');
        expect(typeof title.cover).toBe('string');
      }
    });

    it('Should have the expected keys in the data object', () => {
      for (const title of titles) {
        expect(title).toHaveProperty('title');
        expect(title).toHaveProperty('slug');
        expect(title).toHaveProperty('type');
        expect(title).toHaveProperty('cover');
      }
    });
  });

  describe('Fixture fake', () => {
    const titles = getTitlesMobile(MobileFake);

    it('Should return empty strings for slug & cover if not found', () => {
      // @ts-expect-error: Object is possibly 'undefined'.
      expect(titles[0].slug).toBe('');
      // @ts-expect-error: Object is possibly 'undefined'.
      expect(titles[0].cover).toBe('');
    });
  });
});

describe('getPages()', () => {
  describe('Fixture browse', () => {
    const pages = getPages(MobileBrowse);

    it('Returns an array containing the pages', () => {
      expect(Array.isArray(pages)).toBe(true);
      expect(pages).toHaveLength(11);
    });

    it('should return an array of unique pages', () => {
      const uniquePages = Array.from(new Set(pages));

      expect(uniquePages).toEqual(pages);
    });
  });

  describe('Fixture browse-query', () => {
    const pages = getPages(MobileBQ);

    it('Returns an array containing the pages', () => {
      expect(Array.isArray(pages)).toBe(true);
      expect(pages).toHaveLength(1);
    });

    it('should return an array of unique pages', () => {
      const uniquePages = Array.from(new Set(pages));

      expect(uniquePages).toEqual(pages);
    });
  });

  describe('Fixture browse-query-page-10', () => {
    const pages = getPages(MobileBQPT);

    it('Returns an array containing the pages', () => {
      expect(Array.isArray(pages)).toBe(true);
      expect(pages).toHaveLength(11);
    });

    it('should return an array of unique pages', () => {
      const uniquePages = Array.from(new Set(pages));

      expect(uniquePages).toEqual(pages);
    });
  });
});
