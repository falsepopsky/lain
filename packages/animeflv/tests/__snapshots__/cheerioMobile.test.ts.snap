// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getTitlesMobile() Fixture browse Should return titles 1`] = `
[
  {
    "cover": "/uploads/animes/covers/3821.jpg",
    "slug": "/anime/ayaka",
    "title": "Ayaka",
    "type": "Anime",
  },
  {
    "cover": "/uploads/animes/covers/3820.jpg",
    "slug": "/anime/ryza-no-atelier-tokoyami-no-joou-to-himitsu-no-kakurega",
    "title": "Ryza no Atelier: Tokoyami no Joou to Himitsu no Kakurega",
    "type": "Anime",
  },
  {
    "cover": "/uploads/animes/covers/3822.jpg",
    "slug": "/anime/uchi-no-kaisha-no-chiisai-senpai-no-hanashi",
    "title": "Uchi no Kaisha no Chiisai Senpai no Hanashi",
    "type": "Anime",
  },
  {
    "cover": "/uploads/animes/covers/3819.jpg",
    "slug": "/anime/horimiya-piece",
    "title": "Horimiya: Piece",
    "type": "Anime",
  },
  {
    "cover": "/uploads/animes/covers/3818.jpg",
    "slug": "/anime/bang-dream-its-mygo",
    "title": "BanG Dream! It's MyGO!!!!!",
    "type": "Anime",
  },
  {
    "cover": "/uploads/animes/covers/3817.jpg",
    "slug": "/anime/genjitsu-no-yohane-sunshine-in-the-mirror",
    "title": "Genjitsu no Yohane: Sunshine in the Mirror",
    "type": "Anime",
  },
  {
    "cover": "/uploads/animes/covers/3816.jpg",
    "slug": "/anime/black-clover-mahou-tei-no-ken",
    "title": "Black Clover: Mahou Tei no Ken",
    "type": "Película",
  },
  {
    "cover": "/uploads/animes/covers/3815.jpg",
    "slug": "/anime/5toubun-no-hanayome-movie",
    "title": "5-toubun no Hanayome Movie",
    "type": "Película",
  },
  {
    "cover": "/uploads/animes/covers/3813.jpg",
    "slug": "/anime/kimi-wo-aishita-hitori-no-boku-e",
    "title": "Kimi wo Aishita Hitori no Boku e",
    "type": "Película",
  },
  {
    "cover": "/uploads/animes/covers/3814.jpg",
    "slug": "/anime/boku-ga-aishita-subete-no-kimi-e",
    "title": "Boku ga Aishita Subete no Kimi e",
    "type": "Película",
  },
  {
    "cover": "/uploads/animes/covers/3812.jpg",
    "slug": "/anime/tensei-shitara-slime-datta-ken-movie-guren-no-kizunahen",
    "title": "Tensei shitara Slime Datta Ken Movie: Guren no Kizuna-hen",
    "type": "Película",
  },
  {
    "cover": "/uploads/animes/covers/3811.jpg",
    "slug": "/anime/niehime-to-kemono-no-ou",
    "title": "Niehime to Kemono no Ou",
    "type": "Anime",
  },
  {
    "cover": "/uploads/animes/covers/3810.jpg",
    "slug": "/anime/watashi-ni-tenshi-ga-maiorita-precious-friends",
    "title": "Watashi ni Tenshi ga Maiorita! Precious Friends",
    "type": "Película",
  },
  {
    "cover": "/uploads/animes/covers/3809.jpg",
    "slug": "/anime/ousama-ranking-yuuki-no-takarabako",
    "title": "Ousama Ranking: Yuuki no Takarabako",
    "type": "Anime",
  },
  {
    "cover": "/uploads/animes/covers/3806.jpg",
    "slug": "/anime/majutsushi-orphen-hagure-tabi-seiikihen",
    "title": "Majutsushi Orphen Hagure Tabi: Seiiki-hen",
    "type": "Anime",
  },
  {
    "cover": "/uploads/animes/covers/3808.jpg",
    "slug": "/anime/oshi-no-ko",
    "title": "[Oshi No Ko]",
    "type": "Anime",
  },
  {
    "cover": "/uploads/animes/covers/3807.jpg",
    "slug": "/anime/the-idolmster-cinderella-girls-u149",
    "title": "The IDOLM@STER Cinderella Girls: U149",
    "type": "Anime",
  },
  {
    "cover": "/uploads/animes/covers/3805.jpg",
    "slug": "/anime/the-marginal-service",
    "title": "The Marginal Service",
    "type": "Anime",
  },
  {
    "cover": "/uploads/animes/covers/3803.jpg",
    "slug": "/anime/dead-mount-death-play",
    "title": "Dead Mount Death Play",
    "type": "Anime",
  },
  {
    "cover": "/uploads/animes/covers/3804.jpg",
    "slug": "/anime/kimi-wa-houkago-insomnia",
    "title": "Kimi wa Houkago Insomnia",
    "type": "Anime",
  },
  {
    "cover": "/uploads/animes/covers/3802.jpg",
    "slug": "/anime/kanojo-ga-koushakutei-ni-itta-riyuu",
    "title": "Kanojo ga Koushaku-tei ni Itta Riyuu",
    "type": "Anime",
  },
  {
    "cover": "/uploads/animes/covers/3801.jpg",
    "slug": "/anime/kimetsu-no-yaiba-katanakaji-no-satohen",
    "title": "Kimetsu no Yaiba: Katanakaji no Sato-hen",
    "type": "Anime",
  },
  {
    "cover": "/uploads/animes/covers/3800.jpg",
    "slug": "/anime/world-dai-star",
    "title": "World Dai Star",
    "type": "Anime",
  },
  {
    "cover": "/uploads/animes/covers/3799.jpg",
    "slug": "/anime/isekai-shoukan-wa-nidome-desu",
    "title": "Isekai Shoukan wa Nidome desu",
    "type": "Anime",
  },
]
`;

exports[`getTitlesMobile() Fixture browse-query Should return titles 1`] = `
[
  {
    "cover": "/uploads/animes/covers/752.jpg",
    "slug": "/anime/saint-seiya-omega",
    "title": "Saint Seiya Omega",
    "type": "Anime",
  },
  {
    "cover": "/uploads/animes/covers/1573.jpg",
    "slug": "/anime/saint_seiya_legend_of_sanctuary",
    "title": "Saint Seiya: Legend of Sanctuary",
    "type": "Película",
  },
  {
    "cover": "/uploads/animes/covers/330.jpg",
    "slug": "/anime/saint-seiya-the-lost-canvas",
    "title": "Saint Seiya: The Lost Canvas",
    "type": "Anime",
  },
  {
    "cover": "/uploads/animes/covers/3060.jpg",
    "slug": "/anime/saint-seiya-saintia-shou",
    "title": "Saint Seiya: Saintia Shou",
    "type": "Anime",
  },
  {
    "cover": "/uploads/animes/covers/328.jpg",
    "slug": "/anime/saint-seiya-the-hades-chapter-inferno",
    "title": "Saint Seiya: The Hades Chapter - Inferno",
    "type": "OVA",
  },
  {
    "cover": "/uploads/animes/covers/1966.jpg",
    "slug": "/anime/saint-seiya-soul-of-gold",
    "title": "Saint Seiya: Soul of Gold",
    "type": "Anime",
  },
  {
    "cover": "/uploads/animes/covers/327.jpg",
    "slug": "/anime/saint-seiya-the-hades-chapter-elysion",
    "title": "Saint Seiya: The Hades Chapter - Elysion",
    "type": "OVA",
  },
  {
    "cover": "/uploads/animes/covers/326.jpg",
    "slug": "/anime/saint-seya",
    "title": "Saint Seiya",
    "type": "Anime",
  },
  {
    "cover": "/uploads/animes/covers/329.jpg",
    "slug": "/anime/saint-seiya-the-hades-chapter-sanctuary",
    "title": "Saint Seiya: The Hades Chapter - Sanctuary",
    "type": "OVA",
  },
]
`;

exports[`getTitlesMobile() Fixture browse-query-page-10 Should return titles 1`] = `
[
  {
    "cover": "/uploads/animes/covers/1366.jpg",
    "slug": "/anime/gekijouban-slayers",
    "title": "Gekijouban Slayers",
    "type": "Película",
  },
  {
    "cover": "/uploads/animes/covers/1394.jpg",
    "slug": "/anime/lupin-sansei-vs-meitantei-conan-the-movie",
    "title": "Lupin Sansei vs. Meitantei Conan The Movie",
    "type": "Película",
  },
  {
    "cover": "/uploads/animes/covers/1407.jpg",
    "slug": "/anime/mobile-suit-gundam-the-08th-ms-team",
    "title": "Mobile Suit Gundam: The 08th MS Team",
    "type": "OVA",
  },
  {
    "cover": "/uploads/animes/covers/1577.jpg",
    "slug": "/anime/saiunkoku-monogatari",
    "title": "Saiunkoku Monogatari",
    "type": "Anime",
  },
  {
    "cover": "/uploads/animes/covers/1851.jpg",
    "slug": "/anime/mobile-suit-gundam-0080-war-in-the-pocket",
    "title": "Mobile Suit Gundam 0080: War in the Pocket",
    "type": "OVA",
  },
  {
    "cover": "/uploads/animes/covers/681.jpg",
    "slug": "/anime/megaman-maverick-hunter-x-the-day-of-sigma",
    "title": "MegaMan Maverick Hunter X The Day of Sigma",
    "type": "OVA",
  },
  {
    "cover": "/uploads/animes/covers/384.jpg",
    "slug": "/anime/kami-nomi-zo-shiru-sekai-2",
    "title": "Kami Nomi zo Shiru Sekai 2",
    "type": "Anime",
  },
  {
    "cover": "/uploads/animes/covers/3469.jpg",
    "slug": "/anime/osananajimi-ga-zettai-ni-makenai-love-comedy",
    "title": "Osananajimi ga Zettai ni Makenai Love Comedy",
    "type": "Anime",
  },
  {
    "cover": "/uploads/animes/covers/3672.jpg",
    "slug": "/anime/totsukuni-no-shoujo-2022",
    "title": "Totsukuni no Shoujo (2022)",
    "type": "OVA",
  },
  {
    "cover": "/uploads/animes/covers/1485.jpg",
    "slug": "/anime/tsubasa-chronicle-tori-kago-no-kuni-no-himegimi",
    "title": "Tsubasa Chronicle: Tori Kago no Kuni no Himegimi",
    "type": "Película",
  },
  {
    "cover": "/uploads/animes/covers/1732.jpg",
    "slug": "/anime/detective-conan-movie-18",
    "title": "Detective Conan Movie 18: The Sniper from Another Dimension",
    "type": "Película",
  },
  {
    "cover": "/uploads/animes/covers/2546.jpg",
    "slug": "/anime/danganronpa-3-the-end-of-kibougamine-gakuen-kibou-hen",
    "title": "Danganronpa 3: The End of Kibougamine Gakuen - Kibou-hen",
    "type": "OVA",
  },
  {
    "cover": "/uploads/animes/covers/2771.jpg",
    "slug": "/anime/ballroom-e-youkoso",
    "title": "Ballroom e Youkoso",
    "type": "Anime",
  },
  {
    "cover": "/uploads/animes/covers/2829.jpg",
    "slug": "/anime/mahoutsukai-no-yome",
    "title": "Mahoutsukai no Yome",
    "type": "Anime",
  },
  {
    "cover": "/uploads/animes/covers/392.jpg",
    "slug": "/anime/hidan-no-aria",
    "title": "Hidan no Aria",
    "type": "Anime",
  },
  {
    "cover": "/uploads/animes/covers/3407.jpg",
    "slug": "/anime/sk",
    "title": "SK∞",
    "type": "Anime",
  },
  {
    "cover": "/uploads/animes/covers/1393.jpg",
    "slug": "/anime/the-law-of-ueki",
    "title": "The Law of Ueki",
    "type": "Anime",
  },
  {
    "cover": "/uploads/animes/covers/2117.jpg",
    "slug": "/anime/the-idolmaster-cinderella-girls-2",
    "title": "The IDOLM@STER: Cinderella Girls 2nd Season",
    "type": "Anime",
  },
  {
    "cover": "/uploads/animes/covers/2331.jpg",
    "slug": "/anime/strike-the-blood-valkyria-no-oukoku-hen",
    "title": "Strike the Blood: Valkyria no Oukoku-hen",
    "type": "OVA",
  },
  {
    "cover": "/uploads/animes/covers/2428.jpg",
    "slug": "/anime/concrete-revolutio-choujin-gensou-the-last-song",
    "title": "Concrete Revolutio: Choujin Gensou - The Last Song",
    "type": "Anime",
  },
  {
    "cover": "/uploads/animes/covers/3084.jpg",
    "slug": "/anime/yakusoku-no-neverland",
    "title": "Yakusoku no Neverland",
    "type": "Anime",
  },
  {
    "cover": "/uploads/animes/covers/67.jpg",
    "slug": "/anime/shigofumi",
    "title": "Shigofumi",
    "type": "Anime",
  },
  {
    "cover": "/uploads/animes/covers/292.jpg",
    "slug": "/anime/ghost-in-the-shell-stand-alone-complex",
    "title": "Ghost in the Shell: Stand Alone Complex",
    "type": "Anime",
  },
  {
    "cover": "/uploads/animes/covers/853.jpg",
    "slug": "/anime/shinsekai-yori",
    "title": "Shinsekai yori",
    "type": "Anime",
  },
]
`;
